solutions/platinum/cf-1254D.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: cf-1254D
+source: CF
+title: Tree Queries
+author: Rameez Parwez
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/71594)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O} ((N + Q) \log(N))$
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int N = 150001, mod = 998244353;
+vector<int> adj[N], sum(N);
+vector<int> top(N), in(N), parent(N), depth(N), out(N), sz(N);
```

Comment:
do these need to be fixed values?

Reply:
Yes, since I am using global values that's why I set them to max

Reply:
if ur not gonna resize the vectors then why not just use arrays

Reply:
well, it is just preference ig. 
I use vectors a lot.